float64 delta
float64 gamma
float64 mu
float64 k
---